runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/health/CassandraAsyncHealthCheck.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.quarkus.runtime.internal.health;
+
+import com.datastax.oss.driver.api.core.AsyncPagingIterable;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.quarkus.runtime.api.session.QuarkusCqlSession;
+import io.quarkus.arc.Arc;
+import io.smallrye.health.api.AsyncHealthCheck;
+import io.smallrye.mutiny.Uni;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.util.TypeLiteral;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+@Readiness
+@ApplicationScoped
+public class CassandraAsyncHealthCheck implements AsyncHealthCheck {
+  /** Name of the health-check. */
+  private static final String HEALTH_CHECK_NAME = "DataStax Apache Cassandra Driver health check";
+
+  private static final Type COMPLETION_STAGE_OF_QUARKUS_CQL_SESSION_TYPE =
+      new TypeLiteral<CompletionStage<QuarkusCqlSession>>() {}.getType();
+  static final String HEALTH_CHECK_QUERY =
+      "SELECT data_center, release_version, cluster_name, cql_version FROM system.local";
```

Comment:
Should we take advantage of this PR to change this to what we did for Spring Boot?

Reply:
I was thinking about it but this PR is already introducing a lot of new stuff, so I prefer to do it separately